pkg/chrootarchive/archive.go

```diff
@@ -48,16 +46,20 @@ func untarHandler(tarArchive io.Reader, dest string, options *archive.TarOptions
 		}
 	}
 
-	r := ioutil.NopCloser(tarArchive)
+	r := tarArchive
```

Comment:
What is the reasoning behind this change? I see you removed one defer close to ensure that `Close` is not called twice on that code path, but there are multiple paths into this function as well as 2 definitions of `untarHandler`. See also https://github.com/c0b/docker/blob/fix-race-16924/pkg/archive/archive.go#L746. If this is not necessary to fix the race condition I would separate this sort of cleanup into a different PR.